hello.cpp

```diff
@@ -0,0 +1,46 @@
+#include <thread>
+#include <boost/asio.hpp>
+
+#include "packet.h"
+
+using boost::asio::ip::tcp;
+
+const int port = 3000;
+
+void session(tcp::socket socket, const std::shared_ptr<KeyValueStore>& k) {
+  boost::system::error_code ec;
+  try {
+    for (;;) {
+      char data[Packet::HEADER_LENGTH] = { 0 };
```

Comment:
Not really necessary to create this array every time through the loop. Is having it zeroed every time important? As long as you don't do anything with the data before checking bytes_read you should be okay.